src/AzureStoreFactory.cs

```diff
@@ -102,37 +104,27 @@ public void Delete()
         }
 
         /// <see cref="IStoreFactory.Describe"/>
-        public string Describe() => "[CAS] " + _persist.ServiceClient.BaseUri;
+        public string Describe() => "[CAS] " + _persist.Uri;
 
         /// <summary> Retrieve all accounts that have blobs in stores from this factory. </summary>
         /// <remarks> Accounts are sorted in ascending order. </remarks>
         public async Task<IReadOnlyList<long>> GetAccountsAsync(CancellationToken cancel)
         {
             var accounts = new List<long>();
 
-            var token = default(BlobContinuationToken);
 
-            do
+            var result = await _persist.GetBlobsByHierarchyAsync(
+                traits: BlobTraits.None,
+                states: BlobStates.None,
+                prefix: "",
+                cancellationToken: cancel).ToListAsync().ConfigureAwait(false);
```

Comment:
I believe this `ToListAsync()` is safe, because the number of blobs returned is very small.